java/core/src/main/java/sleeper/core/statestore/AllReferencesToAFile.java

```diff
@@ -29,22 +29,23 @@
 
 /**
  * Reports on all the references for an individual physical file. A file may be referenced in a number of different
- * partitions, and may also have other external references which contribute to a combined reference count (eg. a
- * long-running query may count as a reference to the file). If a file is referenced in multiple partitions, the ranges
- * covered by those partitions must not overlap, or the records in the overlapping portion may be duplicated.
+ * partitions. If a file is referenced in multiple partitions, the ranges covered by those partitions must not overlap,
+ * or the records in the overlapping portion may be duplicated.
  */
 public class AllReferencesToAFile {
 
     private final String filename;
     private final Instant lastStateStoreUpdateTime;
-    private final int totalReferenceCount;
-    private final Map<String, FileReference> internalReferenceByPartitionId;
+    private final Map<String, FileReference> referenceByPartitionId;
+    private final int referenceCount;
 
     private AllReferencesToAFile(Builder builder) {
         filename = Objects.requireNonNull(builder.filename, "filename must not be null");
         lastStateStoreUpdateTime = builder.lastStateStoreUpdateTime;
-        totalReferenceCount = builder.totalReferenceCount;
-        internalReferenceByPartitionId = Objects.requireNonNull(builder.internalReferenceByPartitionId, "internalReferenceByPartitionId must not be null");
+        referenceByPartitionId = Objects.requireNonNull(builder.referenceByPartitionId, "referenceByPartitionId must not be null");
+        referenceCount = referenceByPartitionId.entrySet().stream()
+                .flatMap(entry -> Stream.of(entry.getValue()))
+                .mapToInt(reference -> 1).sum();
```

Comment:
I think this is the same as just calling `referenceByPartitionId.size()`? We probably don't need a separate field, `getReferenceCount` can just call that.